Calling solve! twice  on an op_problem throws an error

**Description**
With a  PowerSimulations.jl operations model calling solve! twice in a row on the operations model throws an error. The error results come not calling build! before the second solve. The error given does not make this very clear. 




**To Reproduce**
```Julia
using Pkg
Pkg.activate(".")
using Revise

using PowerSystems
const PSY = PowerSystems

using JSON
using Xpress
using HiGHS
using PowerSystems
using PowerSimulations
const PSI = PowerSimulations
using PowerSystemCaseBuilder
const PSB = PowerSystemCaseBuilder
using JuMP
using Dates
using DataFrames
using CSV
import WaterModels
using PowerWaterModels


const WM = WaterModels
const PWM = PowerWaterModels


sys_UC = PSB.build_system(PSITestSystems, "c_sys5_uc");

template = ProblemTemplate()
set_network_model!(
    template,
    NetworkModel(DCPPowerModel, duals = [NodalBalanceActiveConstraint], use_slacks = true),
)

set_device_model!(template, Line, StaticBranch)
set_device_model!(template, Transformer2W, StaticBranch)
set_device_model!(template, TapTransformer, StaticBranch)

set_device_model!(template, ThermalStandard, ThermalStandardUnitCommitment)
set_device_model!(template, RenewableDispatch, RenewableFullDispatch)
set_device_model!(template, PowerLoad, StaticPowerLoad)
set_device_model!(template, HydroDispatch, FixedOutput)
set_device_model!(template, HydroEnergyReservoir, HydroDispatchRunOfRiver)
set_device_model!(template, RenewableFix, FixedOutput)

set_service_model!(template, VariableReserve{ReserveUp}, RangeReserve)
set_service_model!(template, VariableReserve{ReserveDown}, RangeReserve)

solver = optimizer_with_attributes(Xpress.Optimizer, "DEFAULTALG"=>2,"PRESOLVE"=>0,"MIPRELSTOP"=>1e-3)

horizon = 24;
interval = Dates.Hour(24);

op_problem = DecisionModel(template, sys_UC; optimizer = solver, horizon = horizon)
build!(op_problem, output_dir = mktempdir())

@time solve!(op_problem)


@time solve!(op_problem)
```

```julia
Error: Decision Problem solve failed
│   exception =
│    AssertionError: !(index in keys(store.optimizer_stats))
│    Stacktrace:
│      [1] write_optimizer_stats!(store::PowerSimulations.DecisionModelStore, stats::PowerSimulations.OptimizerStats, index::DateTime)
│        @ PowerSimulations ~/.julia/packages/PowerSimulations/HKwUO/src/operation/decision_model_store.jl:114
│      [2] macro expansion
│        @ ~/.julia/packages/PowerSimulations/HKwUO/src/operation/decision_model.jl:409 [inlined]
│      [3] macro expansion
│        @ ~/.julia/packages/TimerOutputs/jgSVI/src/TimerOutput.jl:236 [inlined]
│      [4] (::PowerSimulations.var"#77#78"{Bool, Bool, DecisionModel{GenericOpProblem}, Nothing})()
│        @ PowerSimulations ~/.julia/packages/PowerSimulations/HKwUO/src/operation/decision_model.jl:404
│      [5] with_logstate(f::Function, logstate::Any)
│        @ Base.CoreLogging ./logging.jl:511
│      [6] with_logger
│        @ ./logging.jl:623 [inlined]
│      [7] solve!(model::DecisionModel{GenericOpProblem}; export_problem_results::Bool, console_level::Base.CoreLogging.LogLevel, file_level::Base.CoreLogging.LogLevel, disable_timer_outputs::Bool, serialize::Bool, kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
│        @ PowerSimulations ~/.julia/packages/PowerSimulations/HKwUO/src/operation/decision_model.jl:397
│      [8] solve!(model::DecisionModel{GenericOpProblem})
│        @ PowerSimulations ~/.julia/packages/PowerSimulations/HKwUO/src/operation/decision_model.jl:381
│      [9] top-level scope
│        @ ./timing.jl:220
│     [10] include(fname::String)
│        @ Base.MainInclude ./client.jl:451
│     [11] top-level scope
│        @ REPL[65]:1
│     [12] eval
│        @ ./boot.jl:373 [inlined]
│     [13] eval_user_input(ast::Any, backend::REPL.REPLBackend)
│        @ REPL /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/REPL/src/REPL.jl:150
│     [14] repl_backend_loop(backend::REPL.REPLBackend)
│        @ REPL /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/REPL/src/REPL.jl:246
│     [15] start_repl_backend(backend::REPL.REPLBackend, consumer::Any)
│        @ REPL /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/REPL/src/REPL.jl:231
│     [16] run_repl(repl::REPL.AbstractREPL, consumer::Any; backend_on_current_task::Bool)
│        @ REPL /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/REPL/src/REPL.jl:364
│     [17] run_repl(repl::REPL.AbstractREPL, consumer::Any)
│        @ REPL /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/REPL/src/REPL.jl:351
│     [18] (::Base.var"#936#938"{Bool, Bool, Bool})(REPL::Module)
│        @ Base ./client.jl:394
│     [19] #invokelatest#2
│        @ ./essentials.jl:716 [inlined]
│     [20] invokelatest
│        @ ./essentials.jl:714 [inlined]
│     [21] run_main_repl(interactive::Bool, quiet::Bool, banner::Bool, history_file::Bool, color_set::Bool)
│        @ Base ./client.jl:379
│     [22] exec_options(opts::Base.JLOptions)
│        @ Base ./client.jl:309
│     [23] _start()
│        @ Base ./client.jl:495
└ @ PowerSimulations ~/.julia/packages/PowerSimulations/HKwUO/src/operation/decision_model.jl:429
  0.310063 seconds (323.00 k allocations: 14.247 MiB)
RunStatus.FAILED = 2
```

**Expected behavior**
I would expect the error to indicate that build! needs to be called before calling solve!.

--- a/src/operation/decision_model.jl
+++ b/src/operation/decision_model.jl
@@ -333,7 +333,6 @@ function build_model!(model::DecisionModel)
 end
 
 function reset!(model::DecisionModel)
-    # TODO-PJ: This is needed until we remove the ParameterJuMP dependency
     was_built_for_recurrent_solves = built_for_recurrent_solves(model)
     if was_built_for_recurrent_solves
         set_execution_count!(model, 0)

--- a/src/operation/decision_model_store.jl
+++ b/src/operation/decision_model_store.jl
@@ -111,7 +111,9 @@ function write_optimizer_stats!(
     stats::OptimizerStats,
     index::DecisionModelIndexType,
 )
-    @assert !(index in keys(store.optimizer_stats))
+    if index in keys(store.optimizer_stats)
+        @warn "Overwriting optimizer stats"
+    end
     store.optimizer_stats[index] = stats
     return
 end

